code/modules/sanity/sanity_mob.dm

```diff
@@ -310,6 +310,9 @@
 	var/sanity_gain = E.sanity_gain_ingest
 	if(E.id == "ethanol")
 		sanity_gain /= 5
+	else if(istype(E, /datum/reagent/ethanol))
+		var/datum/reagent/ethanol/fine_drink = E
+		sanity_gain *= fine_drink.strength / 5 
```

Comment:
Drink strength is reversed, lower is stronger, beer has 35 while vodka has 5.

Reply:
fixed, now it has a better formula (5 - X/5, X <= 20)